LEAF_Request_Portal/js/form.js

```diff
@@ -281,99 +284,39 @@ var LeafForm = function (containerID) {
       const parentElID =
         event !== null ? parseInt(event.target.id) : parseInt(parID);
 
-      const linkedParentConditions = getConditionsLinkedToParent(parentElID); //get all children directly controlled by this parent, and their ids
-      let uniqueChildIDs = linkedParentConditions.map((c) =>
-        parseInt(c.childIndID)
-      );
-      uniqueChildIDs = Array.from(new Set(uniqueChildIDs));
-
-      let linkedChildConditions = [];
-      uniqueChildIDs.forEach((id) => {
-        linkedChildConditions.push(
-          ...getConditionsLinkedToChild(id, parentElID)
-        ); //get all other possible parents controlling the above children
-      });
-
-      let allConditions = [...linkedParentConditions, ...linkedChildConditions];
-      let hideShowConditions = allConditions.filter((c) =>
-        ["show", "hide"].includes(c.selectedOutcome.toLowerCase())
-      );
-      let prefillConditions = allConditions.filter(
-        c => c.selectedOutcome.toLowerCase() === "pre-fill"
-      );
-      allConditions = [...hideShowConditions, ...prefillConditions]; //orders them so that prefills would be last
-
-      const conditionsByChild = {};
-      allConditions.map((c) => {
-        conditionsByChild[c.childIndID]
-          ? conditionsByChild[c.childIndID].push(c)
-          : (conditionsByChild[c.childIndID] = [c]);
-      });
-
-      setTimeout(() => {
-        //some multiselect updates don't work unless the stack is cleared
-        for (let childID in conditionsByChild) {
-          makeComparisons(childID, conditionsByChild[childID]);
-        }
-      }, 0);
-    };
-    /**
-     *
-     * @param {number} parentID
-     * @returns array of conditions that have the given value for their parentIndID, or empty array
-     */
-    const getConditionsLinkedToParent = (parentID = 0) => {
-      let conditionsLinkedToParent = [];
-      if (parentID !== 0) {
-        for (let entry in formConditionsByChild) {
-          const formConditions = formConditionsByChild[entry].conditions || [];
-          formConditions.forEach((c) => {
-            const formatIsEnabled = allowedChildFormats.some(
-              (f) => f === c.childFormat
-            );
-            //do not include conditions if the recorded condition format (condition.childFormat) does not
-            //match the current format, as this would have unpredictable results
-            if (
-              formConditionsByChild[entry].format === c.childFormat &&
-              formatIsEnabled &&
-              parseInt(c.parentIndID) === parseInt(parentID)
-            ) {
-              conditionsLinkedToParent.push({ ...c });
-            }
-          });
+      //get all conditions for children if any of their controllers match parID. sep by hide/show, prefill
+      let hideShowCondByChild = {};
+      let prefillCondByChild = {};
+      for(let childKey in formConditions) {
+        const id = childKey.slice(2); //id<indID>
```

Comment:
This will reset each iteration (if I tried to redefine it within the same block it would break).  The keys in the formConditions object are the prefix 'id '+ the ID of the indicator, and I need to get only the ID.  I can wrap this though to make certain they do start with 'id', in case unexpected keys somehow end up in there.